### ISSUE TITLE
blackout has no affect

### ISSUE THREAD

**Issue by msupino** [guardian/alerta#228]
Trying to disable notifications for a few minutes from urlmon, alerts are still sent, cant understand why, any clues ?

**Comment by satterly** [comment]
Are you using the blackout periods page in the web UI like this ... ?

<img width="1108" alt="screen shot 2016-07-12 at 22 49 48" src="https://cloud.githubusercontent.com/assets/615057/16784926/20416f80-4883-11e6-88dd-4ce909b6f2a5.png">

If so what settings are you using?

**Comment by msupino** [comment]
tried both with the UI and API/CLI, a line is added to the blackout list, i can see it in the UI, but when an alert that should match the blackout is recevied, an email/slack notifitcation is sent.

**Comment by satterly** [comment]
I've been able to reproduce this problem. I'll need to investigate further.

**Comment by satterly** [comment]
Actually I've only been only to get the plugin `pre_receive()` hook to run when an alert is blacklisted. And this is as designed. The plugin `post_receive()` hook won't run for alerts being blacklisted.

Have you modified the slack / email plugins to include actions in the `pre_receive()` hook?

**Comment by msupino** [comment]
no, i didnt change anything in the plugins

the doc sais that an alert for item in blackout period should get HTTP/202, right ? i see http/200

**Comment by satterly** [comment]
So it sounds like the alert isn't matching the blackout conditions. If you run `alerta send` from the command line you should see  "Suppressed alert during blackout period" like this ...
```
$ alerta send -S Svc -E Development -r r1 -e event6 -C event1 -C event3 -s warning -t 3 -S Network -r r2
03d75d1a-d68b-4e25-819c-e24eb05e4638 (Suppressed alert during blackout period)
```
And the server debug output look like this ...
```
$ alertad --debug
 * Running on http://0.0.0.0:8080/ (Press CTRL+C to quit)
 * Restarting with stat
2016-07-13 08:02:17,394 - alerta.app[97492]: INFO - Starting alerta version 4.1.8 ...
2016-07-13 08:02:17,400 - alerta.app[97492]: INFO - Using MongoDB version 3.2.4 ...
127.0.0.1 - - [13/Jul/2016 08:02:20] "POST /alert HTTP/1.1" 202 -
```

**Comment by msupino** [comment]
from the cli, this works as expected, the message of Suppressed during blackout period is printed

but an alert received from urlmon plugin seems not to behave the same.

**Comment by satterly** [comment]
What are the blackout period filters that you have in place? ie. are you trying to suppress by `service` or `resource` etc

**Comment by msupino** [comment]
tried both, currently by Service.

**Comment by satterly** [comment]
Can we continue on [gitter](https://gitter.im/alerta/chat)?

**Comment by msupino** [comment]
Yes, in two hours is ok?

**Comment by satterly** [comment]
no problem.

**Comment by msupino** [comment]
any update ? maybe this is releated to the "normal" status sent every minute when there is no issue ?

**Comment by satterly** [comment]
I finally got around to trying to reproduce this problem with URLmon and it suppressed the alert as expected for me ...
```
2016-08-02 00:42:55,631 - alerta.urlmon: INFO - Thread-4 check complete.
2016-08-02 00:42:55,631 - alerta.urlmon: DEBUG - Waiting on input queue...
2016-08-02 00:42:56,023 - alerta.urlmon: DEBUG - URL: https://www.statuscake.com/, Status: OK (200), Round-Trip Time: 1419ms -> HttpResponseOK
2016-08-02 00:42:56,025 - requests.packages.urllib3.connectionpool: INFO - Resetting dropped connection: localhost
2016-08-02 00:42:56,032 - requests.packages.urllib3.connectionpool: DEBUG - "POST /alert HTTP/1.1" 202 126
2016-08-02 00:42:56,033 - alerta.api: DEBUG - Request Headers: {'Content-Length': '771', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'User-Agent': 'python-requests/2.10.0', 'Connection': 'keep-alive', 'Content-Type': 'application/json', 'Authorization': 'Key '}
2016-08-02 00:42:56,033 - alerta.api: DEBUG - Request Body: {"status": "unknown", "origin": "alerta-urlmon/macbookpro.home", "resource": "status-cake", "severity": "normal", "correlate": ["HttpConnectionError", "HttpServerError", "HttpClientError", "HttpRedirection", "HttpContentError", "HttpResponseSlow", "HttpResponseOK", "HttpResponseRegexError", "HttpResponseRegexOK"], "tags": ["test"], "text": "HTTP server responded with status code 200 in 1419ms", "createTime": "2016-08-01T23:42:56.023Z", "value": "OK (200)", "event": "HttpResponseOK", "environment": "Production", "customer": null, "service": ["StatusCake"], "rawData": "", "timeout": 86400, "attributes": {"thresholdInfo": "https://www.statuscake.com/ : RT > 5000 RT > 10000 x 1"}, "group": "Web", "type": "serviceAlert", "id": "20e24aa2-82c3-4afc-8f67-debce2a20136"}
2016-08-02 00:42:56,033 - alerta.api: DEBUG - Response Headers: {'Date': 'Mon, 01 Aug 2016 23:42:56 GMT', 'Content-Length': '126', 'Content-Type': 'application/json', 'Server': 'Werkzeug/0.10.4 Python/2.7.9'}
2016-08-02 00:42:56,033 - alerta.api: DEBUG - Response Body: {
  "status": "ok",
  "message": "Suppressed alert during blackout period",
  "id": "74260580-67d6-4bcb-abaf-74b51d995f93"
}
2016-08-02 00:42:56,034 - alerta.urlmon: INFO - Thread-6 check complete.
2016-08-02 00:42:56,034 - alerta.urlmon: DEBUG - Waiting on input queue...
```
Blackouts ...
```
(alerta)macbookpro:~ nsatterl$ alerta --profile development blackout -S StatusCake -E Production
2ab35f9b-d27a-4e7c-b884-0537ebccc358
(alerta)macbookpro:~ nsatterl$ alerta --profile development blackouts
ID       CUSTOMER         ENVIRONMENT      SERVICE          RESOURCE         EVENT            GROUP            TAGS                     STATUS   START               DURATION
afebd3f7 *                Production       *                *                *                Performance      *                        expired  2016/07/12 22:31:00 3600s
a9372f3b *                Development      Network          *                *                *                *                        expired  2016/07/12 22:49:00 3600s
afa9d1b8 *                Development      Network          *                *                *                *                        active   2016/07/12 22:49:00 5878800s
1af87346 *                None             *                *                *                *                *                        active   2016/08/02 00:36:40 3600s
2ab35f9b *                Production       StatusCake       *                *                *                *                        active   2016/08/02 00:36:56 3600s
```
You haven't set `BLACKOUT_ACCEPT` to `normal` in `alertad.conf` have you?

**Comment by msupino** [comment]
there is no value for BLACKOUT_ACCEPT in my config

i am out of the office for a few days, ill check again next week, thanks.

**Comment by msupino** [comment]
i found that the blackout period with a API key that not customer bound will return "blackout period", while a key of a specific customer wont, although the blackout period customer is *

**Comment by satterly** [comment]
@msupino This should now be fixed in alerta server version 4.7.32
